src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -0,0 +1,119 @@
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.InlineDeclarationDiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider() => new InlineDeclarationFixAllProvider(this);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                context.Diagnostics);
+            return SpecializedTasks.EmptyTask;
+        }
+
+        private Task<Document> FixAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+            => FixAllAsync(document, ImmutableArray.Create(diagnostic), cancellationToken);
+
+        private async Task<Document> FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var editor = new SyntaxEditor(root, document.Project.Solution.Workspace);
+            var options = document.Project.Solution.Workspace.Options;
+
+            var useVarWhenDeclaringLocals = options.GetOption(CSharpCodeStyleOptions.UseVarWhenDeclaringLocals);
+            var useImplicitTypeForIntrinsicTypes = options.GetOption(CSharpCodeStyleOptions.UseImplicitTypeForIntrinsicTypes).Value;
+
+            foreach (var diagnostic in diagnostics)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                AddEdits(root, editor, diagnostic, useVarWhenDeclaringLocals, useImplicitTypeForIntrinsicTypes, cancellationToken);
+            }
+
+            var newRoot = editor.GetChangedRoot();
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private void AddEdits(
+            SyntaxNode root, SyntaxEditor editor, Diagnostic diagnostic, 
+            bool useVarWhenDeclaringLocals, bool useImplicitTypeForIntrinsicTypes, 
+            CancellationToken cancellationToken)
+        {
+            var declaratorLocation = diagnostic.AdditionalLocations[0];
+            var identifierLocation = diagnostic.AdditionalLocations[1];
+
+            var declarator = (VariableDeclaratorSyntax)declaratorLocation.FindNode(cancellationToken);
+            var identifier = (IdentifierNameSyntax)identifierLocation.FindNode(cancellationToken);
+
+            var declaration = (VariableDeclarationSyntax)declarator.Parent;
+            if (declaration.Variables.Count == 1)
+            {
+                // Remove the entire declaration statement.
+                editor.RemoveNode(declaration.Parent);
+            }
+            else
+            {
+                // Otherwise, just remove the single declarator.
+                editor.RemoveNode(declarator);
+            }
```

Comment:
In general, we shouldn't delete comments if we can help it. Those sorts of bugs always end up getting reported later. :smile:

Inline Temporary Variable goes through a lot of effort to not delete comments when removing the variable declaration. However, the code there is pretty nasty. Perhaps the right thing to do is this:

1. If the entire variable declaration is being removed, just leave any trailing comment in its place, on the same line.
2. If a declarator is being removed from a multi-variable declaration, copy any inline comments attached to the declarator to the out parameter.

Thoughts?